lib/logqueue.c

```diff
@@ -185,13 +185,13 @@ log_queue_check_items(LogQueue *self, gint *timeout, LogQueuePushNotifyFunc para
     {
       gint64 diff;
       gint new_buckets;
-      GTimeVal now;
+      struct timespec now;
 
-      g_get_current_time(&now);
+      timespec_get(&now, TIME_UTC);
```

Comment:
I would use clock_gettime() if possible. Honestly I didn't even know about timespec_get() and was looking for its implementation when I had to realize this is standard too. But seems to be more recent than clock_gettime() being part of C11, whereas clock_gettime() is from the 90s.

Also, I find it strange that I need to specify a timezone to an epoch based timestamp.